File: app/src/main/java/com/exa/companydemo/base/BaseActivity.java
package com.exa.companydemo.base;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

public abstract class BaseActivity extends AppCompatActivity {
    protected final String TAG = getClass().getSimpleName();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getLayoutId());
        initView();
        initData();
    }

    protected abstract void initData();

    protected abstract void initView();

    protected abstract int getLayoutId();
}

File: launcher/src/main/java/com/gxatek/cockpit/launcher/BaseActivity.kt
package com.gxatek.cockpit.launcher

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.viewbinding.ViewBinding
import com.exa.baselib.utils.L
import com.exa.baselib.utils.SystemBarUtil

/**
 * @Author lsh
 * @Date 2023/6/27 10:44
 * @Description
 */
abstract class BaseActivity<T : ViewBinding> : AppCompatActivity() {
    private lateinit var _binding: T
    protected val binding get() = _binding;

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        L.dd()
        _binding = getViewBinding()
        setContentView(_binding.root)
        initView()
        initData()
    }

    protected abstract fun getViewBinding(): T

    protected abstract fun initView()

    protected abstract fun initData()
}
